src/utils/Subscription.js

```diff
@@ -25,15 +23,13 @@ function initListeners() {
       let isSubscribed = true
       if (next === current) next = current.slice()
       next.push(listener)
-      count++
 
       return function unsubscribe() {
-        if (!isSubscribed || count === 0) return
+        if (!isSubscribed || !current) return
```

Comment:
Should be `current.length === 0`. Any array, even an empty one, is always truthy (so this will always be false).